site/src/pages/tutorials/grpc/blog/implement-create.mdx

```diff
@@ -45,77 +48,110 @@ Let's implement a service method `createBlogPost()` in `BlogService.java`.
-                                       .setModifiedAt(now.toEpochMilli())
-                                       .setCreatedAt(now.toEpochMilli())
-                                       .build();
+                       .setId(id)
+                       .setTitle(request.getTitle())
+                       .setContent(request.getContent())
+                       .setModifiedAt(now.toEpochMilli())
+                       .setCreatedAt(now.toEpochMilli())
+                       .build();
       blogPosts.put(id, updated);
       responseObserver.onNext(updated);
       responseObserver.onCompleted();
     }
   }
   ```
 
-## 2. Implement client-side
-
-This time, let's add a proper method for creating a blog post.
-
-Add a method `createBlogPost()` in `BlogClient.java` to send a request to create a blog post.
-  ```java filename=BlogClient.java
-  import org.slf4j.Logger;
-  import org.slf4j.LoggerFactory;
-
-  class BlogClient {
-    private static final Logger logger = LoggerFactory.getLogger(BlogClient.class);
-    void createBlogPost(String title, String content) {
-      final CreateBlogPostRequest request = 
-        CreateBlogPostRequest.newBuilder()
-                             .setTitle(title)
-                             .setContent(content)
-                             .build();
-      final BlogPost response = client.createBlogPost(request);
-      logger.info("[Create response] Title: {} Content: {}",
-                    response.getTitle(), response.getContent());
-    }
+## 2. Register a ServerExtension
+
+In the previous test code, we've connected directly to the server.
+This time, let's use Armeria's <type://ServerExtension> instead.
+This approach automatically handles set-up and tear-down of a server for testing.
+Now we don't have to invoke the main method to set up a server before running our tests.
+
+1. In the `BlogServiceTest` class, register a <type://ServerExtension> as follows.
+  Note that the service instance is added to the configuration.
+  ```java filename=BlogServiceTest.java
+  import org.junit.jupiter.api.extension.RegisterExtension;
+  import com.linecorp.armeria.server.ServerBuilder;
+  import com.linecorp.armeria.server.grpc.GrpcService;
+  import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+  @RegisterExtension
+  static final ServerExtension server = new ServerExtension() {
+      @Override
+      protected void configure(ServerBuilder sb) throws Exception {
+          sb.service(GrpcService.builder()
+                        // Add the service to the configuration
+                        .addService(new BlogService())
+                        .build());
+      }
+  };
+  ```
+  2. Either remove the test method `hello()` or add the `@Disabled` annotation as follows
+  to prevent it from attempting a direct connection to the server.
+  ```java filename=BlogServiceTest.java
+  @Disabled // Add this
+  @Test
+  void hello() {
```

Comment:
@jrhee17 
I've considered your comment and suggest we either remove or minimize the introduction of the hello() method in the gRPC and Thrift tutorials. Although it's used to demonstrate that the server is connectable, including and then removing this method could distract from the main tutorial content. It might be best to concentrate on the core scenario instead.

I'm trying to simplify the first part of gRPC and Thrift tutorials. Sorry for the back and forth. I'll work on it promptly and push the commit soon.

Reply:
Applied here: 
- [Rearrange and simplify gRPC tutorial](https://github.com/line/armeria/pull/5445/commits/c12e6803cc18b9c231d9597efadf3f8bf7168a75)
- [Rearrange and simplify Thrift tutorial](https://github.com/line/armeria/pull/5445/commits/741e37e12b67229c5a7329456c249d94e98c32c2)